cfgov/v1/tests/models/test_related_posts.py

```diff
@@ -148,6 +149,25 @@ def test_related_posts_blog_limit(self):
         self.assertNotIn('Newsroom', related_posts)
         self.assertNotIn('Events', related_posts)
 
+    def test_related_posts_and_filtering(self):
+        """
+        Tests whether related posts are retrieved if and only if they match ALL
+        of the tags on the calling page.
+        """
+
+        self.block.value['relate_posts'] = True
+        self.block.value['relate_newsroom'] = True
+        self.block.value['relate_events'] = True
```

Comment:
@richaagarwal Ah, and now reading your final comment, I see where the disconnect might've been here:

> I'm pretty sure you need to add some tags to the testing scenario and then check that the pages in the result set all share the tag in question. The relate_x fields are additional filters but you don't need them for this test (or rather, it's arbitrary which one you choose. You could add blog posts, events, or newsroom pages with the relevant tag, or any combination thereof.)

The `setUp` [adds tags to child pages in each of those types](https://github.com/cfpb/cfgov-refresh/blob/00709c6f5e15b9589c8f3adc585cce4be9869a48/cfgov/v1/tests/models/test_related_posts.py#L45-L69).

Without messing with the `setUp`, in this test, I need to have `relate_[type]` be `True` for each type, so that I can show that when `and_filtering` is `True`, posts with only one of the tags are returned. In the test I plan to write with `and_filtering = False`, I could set only one of the `relate_[type]`s to `True.